test/test-agent-stopped.js

```diff
@@ -21,9 +21,14 @@ var http = require('http');
 
 describe('test-agent-stopped', function() {
   var agent;
-  before(function() {
+  before(function(done) {
     delete process.env.GCLOUD_PROJECT;
     agent = require('..').start();
+    // Wait 300ms for agent to fail getting remote project id.
```

Comment:
That would simplify things here. I'd like to keep this PR as close to a mechanical refactoring as possible. I can add the event lifecycle as a follow on.